### ISSUE TITLE
NPE in CommitStatusUpdater

### ISSUE THREAD

**Issue by hho** [jenkinsci/gitlab-plugin#814]
### Context
- **Gitlab plugin version**: 1.5.8 / 1.5.9
- **Gitlab version**: 11.0.4
- **Jenkins version**: 2.121.2
- **Job type**: Maven

### Logs & Traces

Maven build output:
```
12:48:35 Triggered by Henning Hoefer <a href="http://gitlab.ubuntuvm/test/dummy/merge_requests/11" target="_blank">GitLab Merge Request #11</a>: test/feature/emoji => develop
12:48:35 [EnvInject] - Loading node environment variables.
12:48:35 [EnvInject] - Preparing an environment for the build.
12:48:35 [EnvInject] - Keeping Jenkins system variables.
12:48:35 [EnvInject] - Keeping Jenkins build variables.
12:48:35 [EnvInject] - Evaluating the Groovy script content
12:48:36 [EnvInject] - Injecting contributions.
12:48:36 Building in workspace /var/lib/jenkins/workspace/jenkins-dummy-merge-requests
12:48:36 ERROR: A Maven installation needs to be available for this project to be built.Either your server has no Maven installations defined, or the requested Maven version does not exist.
12:48:36 Failed to get Gitlab Build list to update status: A Maven installation needs to be available for this project to be built.Either your server has no Maven installations defined, or the requested Maven version does not exist.
12:48:36 ERROR: Build step failed with exception
12:48:36 java.lang.NullPointerException
12:48:36 	at com.dabsquared.gitlabjenkins.util.CommitStatusUpdater.updateCommitStatus(CommitStatusUpdater.java:68)
12:48:36 	at com.dabsquared.gitlabjenkins.util.CommitStatusUpdater.updateCommitStatus(CommitStatusUpdater.java:97)
12:48:36 	at com.dabsquared.gitlabjenkins.publisher.GitLabCommitStatusPublisher.perform(GitLabCommitStatusPublisher.java:57)
12:48:36 	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
12:48:36 	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:744)
12:48:36 	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:690)
12:48:36 	at hudson.maven.MavenModuleSetBuild$MavenModuleSetBuildExecution.post2(MavenModuleSetBuild.java:1073)
12:48:36 	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:635)
12:48:36 	at hudson.model.Run.execute(Run.java:1823)
12:48:36 	at hudson.maven.MavenModuleSetBuild.run(MavenModuleSetBuild.java:543)
12:48:36 	at hudson.model.ResourceController.execute(ResourceController.java:97)
12:48:36 	at hudson.model.Executor.run(Executor.java:429)
12:48:36 Build step 'Publish build status to GitLab' marked build as failure
12:48:36 Finished: FAILURE
```

### Problem description

When a build can't be started (e.g. Maven installation removed for a Maven build), the `CommitStatusUpdater` causes a `NullPointerException` in line 68.

https://github.com/jenkinsci/gitlab-plugin/blob/1cae6192ce83a9b32df0c04e3e161889c46c3e82/src/main/java/com/dabsquared/gitlabjenkins/util/CommitStatusUpdater.java#L59-L68

After the exception is caught in Line 63 the method should probably return – as it can't do anything useful anymore then. I haven't dug in to why `retrieveGitlabProjectIds(…)` re-throws the exception from the build though – maybe that's the actual issue here?

**Comment by shivanshsharma13** [comment]
Hi, If that issue still exists, I would like to work on it. If anyone have any more insights or research on it, that would be a great help for me.

**Comment by krisstern** [comment]
Closing this issue as the problem should have been fixed with Release 1.7.
